### ISSUE TITLE
New model editions (GPT4)

### ISSUE THREAD

**Issue by deep-diver** [tloen/alpaca-lora#340]
Hi @tloen 

I have trained the following models on GPT4 generated Alpaca dataset(from the one in this repo), and they are available through Hugging Face Model hub.
- [chansung/gpt4-alpaca-lora-7b](https://huggingface.co/chansung/gpt4-alpaca-lora-7b)
- [chansung/gpt4-alpaca-lora-13b](https://huggingface.co/chansung/gpt4-alpaca-lora-13b)
- [chansung/gpt4-alpaca-lora-30b](https://huggingface.co/chansung/gpt4-alpaca-lora-30b)

You can also find out the link for the training logs on each Model repository.
I hope this might be useful for someone, and I also hope these could be included in the list in this repo.

**Comment by T-Atlas** [comment]
Hi @deep-diver 
I tried using GPT-4 data to train the adapter myself, but I found compared to models trained with original data, adapter models trained with GPT-4 data will output `instructions` and `inputs` during generation. 
`python generate.py     --load_8bit     --base_model 'decapoda-research/llama-7b-hf'     --lora_weights 'gpt4-alpaca-
lora-7b'`
I want to know if this is normal?
![image](https://github.com/tloen/alpaca-lora/assets/69239617/e641ca46-582a-430a-a493-4e44d1c95bdd)
The following is an example of raw data training
`python generate.py --load_8bit --base_model='decapoda-research/llama-7b-hf'`
![image](https://github.com/tloen/alpaca-lora/assets/69239617/c497e6ab-8c83-4b9c-b866-beb9b84a6129)

**Comment by deep-diver** [comment]
I think so. You need to trim to get after Response

**Comment by T-Atlas** [comment]
> I think so. You need to trim to get after Response

But I think the format and prompt template of these two pieces of data are the same. Do you have any understanding of why there is such a difference?

**Comment by su-park** [comment]
Hello.
I am experiencing the same issue as the one @T-Atlas posted above.
I have prepared a benchmark set and compared the performance of Alpaca-7b with regards to the same prompt.
The instruction and inputs are attached to the generated output like an echo.

**Comment by JianqiaoLu** [comment]
> I think so. You need to trim to get after Response



> > I think so. You need to trim to get after Response
> 
> But I think the format and prompt template of these two pieces of data are the same. Do you have any understanding of why there is such a difference?

It looks like the loss not only applies to model genereated output but also such template such as "instruction:" and "input：{input}"

**Comment by T-Atlas** [comment]
> > I think so. You need to trim to get after Response
> 
> > > I think so. You need to trim to get after Response
> > 
> > 
> > But I think the format and prompt template of these two pieces of data are the same. Do you have any understanding of why there is such a difference?
> 
> It looks like the loss not only applies to model genereated output but also such template such as "instruction:" and "input：{input}"

Sounds reasonable, do you have any attempts to correct it?

**Comment by JianqiaoLu** [comment]
> > > I think so. You need to trim to get after Response
> > 
> > 
> > > > I think so. You need to trim to get after Response
> > > 
> > > 
> > > But I think the format and prompt template of these two pieces of data are the same. Do you have any understanding of why there is such a difference?
> > 
> > 
> > It looks like the loss not only applies to model genereated output but also such template such as "instruction:" and "input：{input}"
> 
> Sounds reasonable, do you have any attempts to correct it?

The only way that comes to my mind is to re fine-tune the model and set labels of "instruction, input, etc" to -100.
